orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/StartTaskHandler.kt

```diff
@@ -39,22 +44,41 @@ class StartTaskHandler(
   override val stageDefinitionBuilderFactory: StageDefinitionBuilderFactory,
   @Qualifier("queueEventPublisher") private val publisher: ApplicationEventPublisher,
   private val taskResolver: TaskResolver,
-  private val clock: Clock
+  private val clock: Clock,
+  private val environment: Environment
 ) : OrcaMessageHandler<StartTask>, ExpressionAware {
 
   override fun handle(message: StartTask) {
     message.withTask { stage, task ->
-      task.status = RUNNING
-      task.startTime = clock.millis()
-      val mergedContextStage = stage.withMergedContext()
-      repository.storeStage(mergedContextStage)
+      if (isTaskEnabled(task)) {
+        task.status = RUNNING
+        task.startTime = clock.millis()
+        val mergedContextStage = stage.withMergedContext()
+        repository.storeStage(mergedContextStage)
 
-      queue.push(RunTask(message, task.id, task.type))
+        queue.push(RunTask(message, task.id, task.type))
+        publisher.publishEvent(TaskStarted(this, mergedContextStage, task))
+      } else {
+        log.debug("Skipping task.id=${task.id} with task.type=${task.type} because ${getTaskToggle(task)}=false")
+        task.status = SKIPPED
+        val mergedContextStage = stage.withMergedContext()
+        repository.storeStage(mergedContextStage)
 
-      publisher.publishEvent(TaskStarted(this, mergedContextStage, task))
+        queue.push(CompleteTask(message, SKIPPED))
+        publisher.publishEvent(TaskComplete(this, mergedContextStage, task))
+      }
     }
   }
 
+  fun getTaskToggle(task: TaskExecution): String {
+    val justTheClassName = task.implementingClass.substring(task.implementingClass.lastIndexOf('.') + 1)
+    val snakifiedClassName = CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_HYPHEN, justTheClassName)
+    return "tasks.$snakifiedClassName.enabled"
```

Comment:
Since the task is still being resolved from the `TaskResolver` I feel like we should be literal with the name here and call this something like `tasks.$snakifiedClassName.skip.enabled`.

Reply:
I don't think we need to snakeify things ... would be nice to at least stay consistent with existing configuration under `tasks`.

```
tasks:
  monitorFront50Task:
    successThreshold: 0
  monitorKatoTask:
    taskNotFoundTimeoutMs: 600000
    terminalRetries:
      enabled: true
```